### PR TITLE
Updated restore docs

### LINKED ISSUE #2333: `influx restore` can't restore to an existing bucket
Our docs for `influx restore` show an example of restoring to an existing bucket. This doesn't actually work at the moment. The restriction is consistent with 1.x, which didn't allow users to restore to an existing DB.

##### Relevant URLs
- Original user report: influxdata/influxdb#20758
- Example in 2.x docs: https://docs.influxdata.com/influxdb/v2.0/reference/cli/influx/restore/#restore-backup-data-to-an-existing-bucket
- Parallel 1.x docs showing the restriction there: https://docs.influxdata.com/influxdb/v1.8/administration/backup_and_restore/#restore-examples

### LINKED ISSUE #2335: Document that `influx restore --full` overwrites the token created by `influx setup`
`influx restore --full` overwrites the _entire_ KV metadata store on the target server, including the auth token created for the initial user via `influx setup`. This can cause confusing failures when users attempt to restore a full backup to a server that was set up with a different initial token than was used by the original backed-up server. The failure path is:

1. User runs `influx setup` to initialize `server1`
   * Server-side of setup generates an initial auth token "ABC"
   * CLI-side of setup writes "ABC" into configs file
2. After some use, user runs `influx backup` to make a local copy of all data on `server1`
3. User runs `influx setup` to initialize `server2` (either a brand-new server, or a redo of `server1`)
   * Server-side of setup generates an initial auth token "DEF"
   * CLI-side of setup writes "DEF" into configs file
4. Users runs `influx restore --full` to upload the backup from `server1`
   * 1st API call of `restore` passes "DEF" as the auth token, and overwrites the entire KV store in `server2`
   * 2nd API call of `restore` attempts to upload a shard of data, passing "DEF" as the auth token. `server2` rejects the API call because its metadata now stores "ABC" as the expected token.
5. All subsequent API/CLI calls using "DEF" as the auth token also fail, for the same reason

The work-around for this is to pass `-t` when running `influx setup` to ensure the 2nd server uses the same initial auth token as the 1st server.

##### Relevant URLs
- OSS bug report: influxdata/influxdb#20749

### PR DESCRIPTION
**Pull request by sanderson**
Closes #2333
Closes #2335

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/backup-restore/restore.md [hunk 1]
```diff
@@ -20,25 +20,103 @@ Use the `influx restore` command to restore backup data and metadata from Influx
...
+
+When restoring data from a backup file set, InfluxDB temporarily moves existing data and metadata.
+Once the process completes, InfluxDB deletes the temporary data.
 If the restore process fails, InfluxDB preserves the data in the temporary location.
 _See [Recover from a failed restore](#recover-from-a-failed-restore)._
 
+{{% note %}}
+#### Cannot restore to existing buckets
+The `influx restore` command cannot restore data to existing buckets.
+Use the `--new-bucket` flag to specify the new bucket to restore data to.
```
Comment (danxmoran) [inline content/influxdb/v2.0/backup-restore/restore.md:34]:
```suggestion
Use the `--new-bucket` flag to create a bucket with a new name and restore data into it.
```

File: content/influxdb/v2.0/backup-restore/restore.md [hunk 2]
```diff
@@ -20,25 +20,103 @@ Use the `influx restore` command to restore backup data and metadata from Influx
...
+
+# OR
+
+influx restore \
+  --input /backups/2020-01-20_12-00/ \
+  --bucket-id 000000000000
+```
+
+If a bucket with the same name as the backed up bucket already exists, use the
+`--new-bucket` flag to specify a new bucket to restore the data to.
```
Comment (danxmoran) [inline content/influxdb/v2.0/backup-restore/restore.md:79]:
```suggestion
`--new-bucket` flag to create a new bucket with a different name and restore data into it.
```

File: content/influxdb/v2.0/backup-restore/restore.md [hunk 3]
```diff
@@ -20,25 +20,103 @@ Use the `influx restore` command to restore backup data and metadata from Influx
...
+    ```sh
+    influx setup --token My5uP3rSecR37t0keN
+    ```
+3. Restore the backup to the new server.
+
+    ```sh
+    influx restore \
+      --input /backups/2020-01-20_12-00/ \
+      --full
+    ```
```
Comment (danxmoran) [inline content/influxdb/v2.0/backup-restore/restore.md:118]: Should we add a "if you don't do this, the restore process will fail in XYZ way..." section below this explaining why?

File: content/influxdb/v2.0/reference/cli/influx/restore/index.md [hunk 4]
```diff
@@ -25,6 +26,14 @@ If the restore process fails, InfluxDB preserves the data in the temporary locat
 _For information about recovering from a failed restore process, see
 [Restore data](/influxdb/v2.0/backup-restore/restore/#recover-from-a-failed-restore)._
 
+{{% note %}}
+#### Cannot restore to existing buckets
+The `influx restore` command cannot restore data to existing buckets.
+Use the `--new-bucket` flag to specify the new bucket to restore data to.
```
Comment (danxmoran) [inline content/influxdb/v2.0/reference/cli/influx/restore/index.md:32]:
```suggestion
Use the `--new-bucket` flag to create a bucket with a new name and restore data into it.
```

File: content/influxdb/v2.0/backup-restore/restore.md (hunk 3 above)
Comment (sanderson) [reply content/influxdb/v2.0/backup-restore/restore.md:118]: Sure, I can add that.

**Review by danxmoran — APPROVED** [review]

**Review by kelseiv — APPROVED** [review]

File: content/influxdb/v2.0/backup-restore/restore.md [hunk 5]
```diff
@@ -20,25 +20,114 @@ Use the `influx restore` command to restore backup data and metadata from Influx
...
+Once the process completes, InfluxDB deletes the temporary data.
 If the restore process fails, InfluxDB preserves the data in the temporary location.
 _See [Recover from a failed restore](#recover-from-a-failed-restore)._
 
+{{% note %}}
+#### Cannot restore to existing buckets
+The `influx restore` command cannot restore data to existing buckets.
+Use the `--new-bucket` flag to create a bucket with a new name and restore data into it.
+To restore data and retain bucket names, [delete existing buckets](/influxdb/v2.0/organizations/buckets/delete-bucket/)
+and then begin the restore process.
```
[the remark this answers is not in the document]
Comment (sanderson) [reply content/influxdb/v2.0/backup-restore/restore.md:36]: No, the wouldn't.

File: content/influxdb/v2.0/backup-restore/restore.md [hunk 6]
```diff
@@ -20,25 +20,114 @@ Use the `influx restore` command to restore backup data and metadata from Influx
...
+
+- backup directory path
 
 ```sh
-# Syntax
-influx restore <path-to-backup-directory>
+influx restore \
+  --input /backups/2020-01-20_12-00/
+```
+
```
[the remark this answers is not in the document]
Comment (sanderson) [reply content/influxdb/v2.0/backup-restore/restore.md:58]: No, this just takes a backup directory and imports the data back into InfluxDB.

File: content/influxdb/v2.0/backup-restore/restore.md [hunk 7]
```diff
@@ -20,25 +20,114 @@ Use the `influx restore` command to restore backup data and metadata from Influx
...
+### Restore data from a specific bucket
+To restore data from a specific bucket in a backup directory, provide the following:
 
-# Example
-influx restore ~/backups/2020-01-20_12-00/
+- backup directory path
```
[the remark this answers is not in the document]
Comment (sanderson) [reply content/influxdb/v2.0/backup-restore/restore.md:62]: A backup directory contains data for one or more buckets.

**Pull request merged**